### DISCUSSION TITLE
Fooocus Prompt in Filename? More than 32?

### DISCUSSION THREAD

**Discussion by AugementedOwl** [lllyasviel/Fooocus#1029] [Q&A]
Is there any way of modifying either of these options? I like to create large numbers of randomised images but so far Automatic is the best option for that because it both outputs with names and can be set to continue infinitely. Best I can see Fooocus can't do either of these things much to my annoyance. 

Does anyone know a way of forcing more than 32 images to be made? 

Or even better, putting part of the prompt in the image name? 

I love Fooocus for general generations but I always seem to have to go back to Automatic sometimes.

**Comment by Overtime7403** [comment]
Fair warning this could break your install. 
I am a complete novice and just a dabbler. You follow ANY of my methods at your own risk!

I have found that if I edit ~/Fooocus/modules/config.py I can achieve a default higher than 32. 
In my config this begins at line 246.
The lines you are looking for appear as follows

> default_image_number = get_config_item_or_set_default(
>     key='default_image_number',
>     default_value=2,
>     validator=lambda x: isinstance(x, int) and 1 <= x <= 2
> )

Change the two values that in my code happen to be set at 2 to what ever number you like. Understand that it may drastically increase resource usage and may not be a good idea.

Further on in the code you can find the following for changing / adding resolutions.
For me it begins at line 270
>available_aspect_ratios = get_config_item_or_set_default(
>    key='available_aspect_ratios',
>    default_value=[
>        '704408', '704*1344', '768*1344', '768*1280', '832*1216', '832*1152',
>        '896*1152', '896*1088', '960*1088', '960*1024', '1024*1024', '1024*960',
>        '1088*960', '1088*896', '1152*896', '1152*832', '1216*832', '1280*768',
>        '1344*768', '1344*704', '1408*704', '1472*704', '1536*640', '1600*640',
>        '1664*576', '1728*576', '1920*1200',
>        '3000*2000', '2160*1440', '1152*768', '1440*960', '1920*1280', '960*640', '480*320', 
>        '1920*1080', '1792*1008', '1664*936', '1536*864', '1408*792', '1280*720', 
>        '1152*648', '1024*576', '896*504', 
>        '1080*1920', '1008*1792', '936*1664', '864*1536', '792*1408', '720*1280',
>        '648*1152', '576*1024', '504*896', '960*540', '540*960', '1500*1000'`

AGAIN USE ANY OF MY METHODS AT YOUR OWN RISK

I have not yet found out how to modify the output file names.

**Comment by mashb1t** [comment]
https://github.com/lllyasviel/Fooocus/pull/1616 has recently been merged, you can set the maximum now in your config.
